src/cli.js

```diff
@@ -1,103 +1,196 @@
-    .example('$0 src/styles')
-    .example('$0 src -o dist')
-    .example('$0 -p styles/**/*.icss -w')
-    .detectLocale(false)
-    .demand(['_'])
-    .alias('c', 'camelCase').describe('c', 'Convert CSS class tokens to camelcase')
-    .alias('o', 'outDir').describe('o', 'Output directory')
-    .alias('p', 'pattern').describe('p', 'Glob pattern with css files')
-    .alias('w', 'watch').describe('w', 'Watch input directory\'s css files or pattern').boolean('w')
-    .alias('d', 'dropExtension').describe('d', 'Drop the input files extension').boolean('d')
-    .alias('h', 'help').help('h')
-    .version(() => require('../package.json').version)
-let argv = yarg.argv;
-let creator;
-
-function writeFile(f) {
-    getSource(f).then((content) => {
-        creator.create(f, content, !!argv.w)
-        .then(content => content.writeFile())
-        .then(content => {
-            console.log('Wrote ' + chalk.green(content.outputFilePath));
-            content.messageList.forEach(message => {
-                console.warn(chalk.yellow('[Warn] ' + message));
-            });
-        })
-        .catch(reason => console.error(chalk.red('[Error] ' + reason)));
-    });
-};
+const readSass = (file, relativeTo) => (
+    new Promise((resolve, reject) => {
+        sass.render(
+            { file },
+            (err, result) => {
+                if (err && (relativeTo && relativeTo !== '/')) {
+                    return resolve([]);
+                } else if (err && (!relativeTo || relativeTo === '/')) {
+                    return reject(err);
+                }
+                return resolve(result.css.toString());
+            },
+        );
+    })
+);
 
-function getSource(file, relativeTo) {
-    return new Promise((resolve, reject) => {
-      sass.render({file}, (err, result) => {
-        if (err && relativeTo && relativeTo !== '/') {
-          return resolve([]);
+const createTypings = (f, creator, cache, handleError, handleWarning, verbose) => (
+    readSass(f)
+    .catch((reason) => {
+        handleError(`${chalk.red(f)}\n${reason}\n`);
+    })
+    .then(content => creator.create(f, content, cache))
+    .then(c => c.writeFile())
+    .then((c) => {
+        if (verbose) {
+            console.info(`Created ${chalk.green(c.outputFilePath)}`);
         }
-  
-        if (err && (!relativeTo || relativeTo === '/')) {
-          return reject(err);
+        c.messageList.forEach((message) => {
+            handleWarning(`${chalk.yellow(f)}\nWarning: ${message}\n`);
+        });
+        return c;
+    })
+    .catch((reason) => {
+        handleError(`${chalk.red(f)}\nError: ${reason}\n`);
+    })
+);
+
+
+const createTypingsForFileOnWatch = (creator, cache, verbose) => (f) => {
+    let warnings = [];
+    let errors = [];
+
+    const cleanUp = () => {
+        warnings.forEach(m => console.warn(m));
+        errors.forEach(e => console.error(e));
```

Comment:
I don't think it's the right place to output messages in cleanUp. If the whole process took a few seconds, user won't see any message until the whole process finishes. I would show messages as soon as possible and you can do in handleError/handleWarning.